src/modules/sub-items/components/sub-items-list/instant.filter.component.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+const FILTER_TIMEOUT = 200;
+
+const InstantFilter = (props) => {
+    const _refInstantFilter = useRef(null);
+    const [filterQuery, setFilterQuery] = useState('');
+    const [itemsMap, setItemsMap] = useState([]);
+    const [filterTimeout, setFilterTimeout] = useState('null');
+
+    useEffect(() => {
+        const items = [..._refInstantFilter.current.querySelectorAll(`.${props.itemClass}`)];
+        const itemsMap = items.reduce(
+            (total, item) => [
+                ...total,
+                {
+                    label: item.textContent.toLowerCase(),
+                    element: item,
+                },
+            ],
+            []
+        );
+
+        setItemsMap(itemsMap);
+    }, []);
+
+    useEffect(() => {
+        window.clearTimeout(filterTimeout);
```

Comment:
`clearTimeout` should go to return function insteaed of here
```suggestion
      useEffect(() => {
        const filterActionTimeout = window.setTimeout(() => {
            const results = itemsMap.filter((item) => item.label.includes(filterQuery.toLowerCase()));

            itemsMap.forEach((item) => item.element.setAttribute('hidden', true));
            results.forEach((item) => item.element.removeAttribute('hidden'));
        }, FILTER_TIMEOUT);

        return () => {
          window.clearTimeout(filterActionTimeout);
        }
    }, [filterQuery]);
```